distgen/generator.py

```diff
@@ -195,6 +196,30 @@ def _recursive_render_spec(self, s, max_passes=25, **kwargs):
 
         return s
 
+    def _enhanced_yaml_module(self, sysconfig):
+
+        # NOTE: This is soo ugly, sorry for that, in future we need to modify
+        # PyYAML to let us specify callbacks, somehow.  But for now, import
+        # yaml right here (local import) to be able to add the
+        # constructors/representers **only** locally (don't modify global
+        # context).
+        def _eval_node(loader, node):
+            return str(eval(str(loader.construct_scalar(node)), {
+                'project': self.project,
+                'config': sysconfig,
+                'macros': sysconfig['macros'],
+            }))
+
+        import yaml
+        try:
+            yaml.add_constructor(u'!eval', _eval_node, yaml.FullLoader)
+            yaml.dg_load = functools.partial(yaml.load, Loader=yaml.FullLoader)
+        except AttributeError:
```

Comment:
Maybe add a comment here that this clause is for older versions of PyYaml that doesnt yet have `Loader=yaml.FullLoader`?

Reply:
I'll do, thanks.